app/assets/javascripts/plugins/filterable-table.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Author: willnigeldejesus@gmail.com
+ * Date:   22th Oct 2024
+ */
+
+(function($) {
+
+  $.fn.filterableTable = function(options) {
```

Comment:
Let me explain the approach here.

Right now I settled with least amount of architecture to make this work. Bulk of the work is going to be done server/rails side instead of here in js. In any case, for further customizations we can pass in `options` when initializing, but we can deal with that once we encounter the need for adding further customization.

So the approach is, for filterable tables, we have a form and the actual table is inside a form under a specific selector id of our  choosing (ex: "#notes-table").
Each filterable table will have by default, hidden fields for `:sort` and `:order`, params that we will use later in filtering/sorting in server side.

Then the simplest approach I could think of here is, we autosubmit when the form changes, make the xhr request, then server will give us an html partial as response and we jquery substitute the contents of '#table-selector' with the response of the server.

So the structure is
#form
   hidden_fields :sort, :order
   form_fields :per_page, :search, :etc
   #table // this is what we replace with server response partial

For sorting, we just attach event listeners for each sortable table-header, which is identifiable by 'data-sort' attr.